app/migrations/Version20240708153845.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Mautic\CoreBundle\Doctrine\AbstractMauticMigration;
+
+final class Version20240708153845 extends AbstractMauticMigration
+{
+    private string $emailStatsTableName;
+
+    protected string $emailsTableName;
+
+    public function preUp(Schema $schema): void
+    {
+        $this->emailStatsTableName = $this->prefix.'email_stats';
+        $this->emailsTableName     = $this->prefix.'emails';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $sql          = sprintf('SELECT id, read_count FROM %s', $this->emailsTableName);
+        $emailsResult = $this->connection->executeQuery($sql)->fetchAllAssociative();
+
+        foreach ($emailsResult as $email) {
+            $statsResult = $this->connection
+              ->executeQuery(
+                  "SELECT email_id, is_read 
+                        FROM {$this->emailStatsTableName} 
+                        WHERE email_id = {$email['id']}"
```

Comment:
Replacing this with a grouped count query, is probably more efficient. 
Email_stats can be huge on some mautic installs.

Reply:
I'm still burnishing my SQL skills. Drupal 8+ didn't require me to write much SQL queries, so I'm working on improving that.

I have updated the PR. I hope this looks good to you.

Reply:
@mallezie could you re-review from the code perspective please? Thanks!